lnclient/lnd/lnd.go

```diff
@@ -313,34 +313,54 @@ func (svc *LNDService) LookupInvoice(ctx context.Context, paymentHash string) (t
 	return transaction, nil
 }
 
+func (svc *LNDService) getPaymentResult(stream routerrpc.Router_SendPaymentV2Client) (*lnrpc.Payment, error) {
+	for {
+		payment, err := stream.Recv()
+		if err != nil {
+			return nil, err
+		}
+
+		if payment.Status != lnrpc.Payment_IN_FLIGHT {
+			return payment, nil
+		}
+	}
+}
+
 func (svc *LNDService) SendPaymentSync(ctx context.Context, payReq string, amount *uint64) (*lnclient.PayInvoiceResponse, error) {
-	sendRequest := &lnrpc.SendRequest{PaymentRequest: payReq}
+	const MAX_PARTIAL_PAYMENTS = 21
```

Comment:
is there a good default provided by LND? this seems way too high to me

Reply:
Sadly no, a good reference can be Polar, where it is set to 16 (I think setting it to a large number would cause more number of in flight HTLCs which can be bad)

https://github.com/jamaljsr/polar/blob/ba6b1cf082732fa8aa12bb6db46c2c57b1a7a089/src/lib/lightning/lnd/lndService.ts#L184

Reply:
I doubt a MPP payment with 16 parts would succeed. The more parts and more hops the more chance of failures. I asked Zeus

Reply:
Atleast in LND, the way it works is it tries to make it in full, if not then it keeps splitting till the min amount depending on the max parts, so we should be good. From https://lightning.engineering/posts/2020-05-07-mpp/

> Initially, LND will try to complete the payment in one part. If that’s not possible, say because channel capacities aren’t sufficient or because other constraints like the fee limit are hit, an attempt is made for half the payment amount. After that HTLC has been launched, the software searches for a second path to pay the remaining amount (the other half). If further failures happen, amounts are repeatedly split until a minimum amount limit is reached or the payment succeeds.

Reply:
I have a feeling (from the blog) that this should be good, reduced to 16 now, let me know what you think

Reply:
Thanks for checking!

Reply:
Evan from Zeus also recommended 16, cool.